### ISSUE TITLE
Bug in example code on front page

### ISSUE THREAD

**Issue by Stannislav** [pytorch/ignite#746]
In the example code in the `README.md` file the statement `model.train()` is outside of the training loop. The function `validate(...)` calls `model.eval()` and the model ends up in eval mode for the rest of the training.

**How it should be:** either put `model.train()` inside the training loop or let `validate(...)` remember and restore the model state.

**Comment by vfdev-5** [comment]
@Stannislav yes, you are right about that. We need to fix this.
